### PR TITLE
fix(#1593): add testId attribute to react components

### PR DESCRIPTION
**Pull request by bdfranck**
**1. What this code change do:**

This change addresses  https://github.com/GovAlta/ui-components/issues/1593 by adding a `testId` property to the following React components: 
- <del>Accordion</del> (I removed it because it is a [breaking change](https://github.com/GovAlta/ui-components/pull/1800#discussion_r1580034789))
- AppHeaderMenu
- Calendar
- DatePicker
- Details
- FileUploadCard
- FileUploadInput
- FooterMetaSection
- SideMenu
- SideMenuGroup
- SideMenuHeading
- Tabs

**2. Make sure that you've checked the boxes below before you submit MR:**

- [x] I have read and followed the [contribution guidelines](../../contributing.md)
- [x] I have run a build locally.
- [x] I have ran unit tests locally.
image

- [x] I have tested the functionality.

### CODE REVIEW

**Comment by bdfranck** [comment]
I fixed a merge conflict, and it created an additional commit. Should I do anything with it? 🤔 

https://github.com/GovAlta/ui-components/pull/1800/commits/22a6b819977eb571f462dcf58d8b4d175f836663

**Comment by lizhuomeng71** [comment]
Issues is verified on Chrome, Firefox, and edge for react and angular, for mobile and desktop site
Screenshot 2024-04-25 114232

File: libs/react-components/src/lib/accordion/accordion.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export interface GoAAccordionProps extends Margins {
   secondaryText?: string;
   heading: string;
   headingContent?: ReactNode;
-  testid?: string;
+  testId?: string;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/accordion/accordion.tsx:29]: This is somewhat of a breaking change. Maybe for now keep the lowercase version (this file only) and in the upcoming Angular major version we can change it to the camelCase version then.

**Comment by bdfranck** [comment]
@chrisolsen Good catch! I removed the Accordion `testid` changes. I also re-did the branch and removed the merge commit. Anything else?

**Comment by bdfranck** [comment]
image

I re-ran my local tests just to make sure.

**Comment by lizhuomeng71** [comment]
Issues is verified on Chrome, Firefox, and edge for react and angular, for mobile and desktop site

**Review by chrisolsen — APPROVED** [review]

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**